openedx/core/djangoapps/catalog/utils.py

```diff
@@ -276,12 +276,16 @@ def get_fulfillable_course_runs_for_entitlement(entitlement, course_runs):
     2) A user can enroll in
     3) A user can upgrade in
     4) Are published
+    5) Are not enrolled in already for an active session
 
     These are the only sessions that can be selected for an entitlement.
     """
 
     enrollable_sessions = []
 
+    enrollments_for_user = CourseEnrollment.enrollments_for_user(entitlement.user).filter(mode=CourseMode.VERIFIED)
```

Comment:
Can you use the entitlements mode here rather than a hard coded CourseMode.Verified?

The hard coded more will not work as an entitlement can be any paid mode.  Verified, Professional, etc.

Reply:
Yup can do :+1: